src/kernel/messagestartchars.h

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_KERNEL_MESSAGESTARTCHARS_H
+#define BITCOIN_KERNEL_MESSAGESTARTCHARS_H
+
+#include <array>
+#include <cstdint>
+
+using MessageStartChars = std::array<uint8_t, 4>;
```

Comment:
Why not `std::byte`?

Reply:
I didn't want to change this data type, since I am not familiar enough with all its users to say what side effects this may have. Also I think `unsigned char` would convey the meaning of the type better than `std::byte`, given that we do use it for string representations.

Reply:
Agreed, this PR has crept enough already :)

(Though std::byte makes sense as a follow-up, considering this will likely end up exposed as part of some api)

Reply:
We already do maths on this in a sense, in that we convert it to an `int32_t` for the `app_id` in wallet/sqlite.cpp. Not seeing what the benefit in switching to `std::byte` here would be.